internal/controllers/nmc_reconciler.go

```diff
@@ -852,8 +853,12 @@ func (p *podManagerImpl) LoaderPodTemplate(ctx context.Context, nmc client.Objec
 	}
 
 	if nms.Config.Modprobe.FirmwarePath != "" {
-		args = append(args, "--"+worker.FlagFirmwareMountPath, worker.FirmwareMountPath)
-		if err = setFirmwareVolume(pod, p.workerCfg.SetFirmwareClassPath); err != nil {
+		firmwareClasspath := p.workerCfg.SetFirmwareClassPath
```

Comment:
one suggestion:
lets move line 837
args = append(args, "--"+worker.FlagFirmwareClassPath, *p.workerCfg.SetFirmwareClassPath)
also into this if clause and completely remove the if clause on line 836

Reply:
Good point. Will do.